### PR TITLE
Alternative HTTP STATUS code update

### PR DESCRIPTION
**Pull request by inoas**
Ref: https://github.com/cakephp/cakephp/pull/9523

### CODE REVIEW

**Comment by inoas** [comment]
Note that these do not exist on IANA list: http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
418 => 'I\'m a Teapot'
499 => 'Client Closed Request',
599 => 'Network Connect Timeout Error',

File: src/Network/Response.php [hunk 1]
```diff
@@ -31,55 +31,100 @@
 class Response
 {
 
+
```
Comment (ADmad) [inline src/Network/Response.php]: Extra blank line

File: src/Network/Response.php [hunk 2]
```diff
@@ -31,55 +31,100 @@
...
     /**
      * Holds HTTP response statuses
-     *
+     * Sources: http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
```
Comment (ADmad) [inline src/Network/Response.php]: You can use `@link` instead of `Sources:`

File: src/Network/Response.php [hunk 3]
```diff
@@ -31,55 +31,100 @@
...
-        422 => 'Unprocessable Entity',
-        429 => 'Too Many Requests',
-        451 => 'Unavailable For Legal Reasons',
-        500 => 'Internal Server Error',
-        501 => 'Not Implemented',
-        502 => 'Bad Gateway',
-        503 => 'Service Unavailable',
-        504 => 'Gateway Time-out',
-        505 => 'Unsupported Version'
+    protected $statusCodes = [
```
Comment (ADmad) [inline src/Network/Response.php]: Don't rename the var, keep it `$_statusCodes`

File: src/Network/Response.php [hunk 4]
```diff
@@ -31,55 +31,100 @@
...
+        // 1×× 1xx: Informational - Request received, continuing process
```
Comment (ADmad) [inline src/Network/Response.php]: Please remove all these inline comments. People can refer to the referenced links if they need more info about each code.

Comment (inoas) [inline src/Network/Response.php]: Could at least the RFC pointers stay? It should be easy to discover then and help devs decide what they really want.

Comment (ADmad) [inline src/Network/Response.php]: > Could at least the RFC pointers stay?

No, it's not CakePHP's job to make devs aware about all the RFCs out there.

**Comment by ADmad** [comment]
Closing as #9523 already takes care of adding the missing codes.

**Comment by inoas** [comment]
Hope you did a diff yourself. As I merged a lot of stuff from different sources.

**Pull request closed without merging**